android/app/src/main/AndroidManifest.xml

```diff
@@ -3,16 +3,24 @@
     package="com.kylekun.one_second_diary">
 
     <uses-permission android:name="android.permission.INTERNET" />
-    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
-    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
+    <uses-permission android:name="android.permission.READ_MEDIA_VIDEO" />
+    <uses-permission android:name="android.permission.READ_MEDIA_IMAGES" />
+    <uses-permission android:name="android.permission.READ_MEDIA_AUDIO" />
+    <uses-permission android:name="android.permission.POST_NOTIFICATIONS"/>
+    <!-- Android 13 introduced new granular permissions. The READ_EXTERNAL_STORAGE permission will only work on Android 12 & below devices -->
+    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" android:maxSdkVersion="32" />
+    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" android:maxSdkVersion="32"/>
+    <uses-permission android:name="android.permission.MANAGE_EXTERNAL_STORAGE" /> 
```

Comment:
Is this permission really needed?

Reply:
I think this would still require the permission, we would be removing files from that directory we created before.

Reply:
> Other apps are also writing in that directory, right?

Yeah, I found that WhatsApp actually writes to this `media` directory on android 11+ I think (I have verified this on android 12 & 13), while it writes to the root directory (like our app is doing currently) on older android versions.

I created a separate local branch some minutes ago where I was able to test migrating the code to work without this extra permission using the same approach. I can't get the files from the previous directory to move over to the new directory without requesting the same permission though, otherwise, it works perfectly.

I think for our options, we could:
- have a second mini-app we can direct the users to (for the sole purpose of restoring previous data)
- have the user copy the files manually (we can provide instructions of some sort)
- roll with this for the time being

It seems Play Store is quite a bit more stringent for apps that require this permission, but otherwise, I think we can do with the current implementation.

I can make any changes if we're all on the same page though :)

Reply:
FWIW, i also just noticed the data in the `media` folder is also deleted when the app is uninstalled, but this doesn't happen when we're working with the custom location

Reply:
Yeah, manual copying is hard for simple users, advanced ones should understand the need of permission + we are open source, I vote for keeping this permission for all android versions to avoid problems.